docs/tutorials/dev_add_test_group.md

```diff
@@ -0,0 +1,2218 @@
+
+Returning to the `default` test case, we are now ready to add
+`initial_state` and `forward` steps to the test case.  In
+`polaris/ocean/tests/baroclinic_channel/default/__init.py`, we add:
+
+```python
+from polaris import TestCase
+from polaris.ocean.tests.baroclinic_channel.forward import Forward
+from polaris.ocean.tests.baroclinic_channel.initial_state import InitialState
+from polaris.ocean.tests.baroclinic_channel.viz import Viz
+
+
+class Default(TestCase):
+    def __init__(self, test_group, resolution):
+        ...
+
+        self.add_step(
+            InitialState(test_case=self, resolution=resolution))
+
+        self.add_step(
+            Forward(test_case=self, ntasks=4, min_tasks=4, openmp_threads=1,
+                    resolution=resolution, run_time_steps=3))
+
+        self.add_step(
+            Viz(test_case=self))
+```
+
+We hard-code the `forward` test case to run on 4 cores and 1 thread, and do
+not pass a viscosity (meaning it will use the default value from
+`forward.yaml`).
+
+### Adding validation
+
+One of the main purposes of having test cases is to validate changes to the
+code.  You can use polaris' validation code to compare the output of different
+steps to one another (or files within a single step), but a very common type
+of validation is to check if the contents of files exactly match the contents
+of the same files from a "baseline" run (performed with a different branch of
+E3SM and/or polaris).
+
+Validation happens at the test-case level so that steps can be compared with
+one another.  Well add baseline validation for both the initial state and
+forward runs:
+
+```python
+from polaris import TestCase
+from polaris.validate import compare_variables
+
+
+class Default(TestCase):
+    def validate(self):
+        """
+        Compare ``temperature``, ``salinity``, ``layerThickness`` and
+        ``normalVelocity`` in the ``forward`` step with a baseline if one was
+        provided.
+        """
+        super().validate()
+
+        variables = ['temperature', 'salinity', 'layerThickness']
+        compare_variables(test_case=self, variables=variables,
+                          filename1='initial_state/initial_state.nc')
+
+        variables = ['temperature', 'salinity', 'layerThickness',
+                     'normalVelocity']
+        compare_variables(test_case=self, variables=variables,
+                          filename1='forward/output.nc')
+```
+We check salinity, temperature and layer thickness in both the initial state
+and forward steps, and the normal velocity only in the forward step (it's
+initialized to zero in the initial state so we feel confident that we can skip
+it, though validating it would certainly do no harm).  Since we only provide
+`filename1` in each call to {py:func}`polaris.validate.compare_variables()`,
+we will only do this validation if a user has set up the test case with a
+baseline, see {ref}`dev-validation`.
+
+### Test things out!
```

Comment:
Now I see this section. As I mentioned before, I prefer to move this up earlier, so that folks can check if their test appears after adding the minimum amount of code necessary.